pkg/controller/direct/resolverefs/projectref.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolverefs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AsProjectRef converts a generic ResourceRef into a ProjectRef
+func AsProjectRef(in *v1alpha1.ResourceRef) *v1beta1.ProjectRef {
+	if in == nil {
+		return nil
+	}
+	return &v1beta1.ProjectRef{
+		Namespace: in.Namespace,
+		Name:      in.Name,
+		External:  in.External,
+		Kind:      in.Kind,
+	}
+}
+
+type Project struct {
+	ProjectID string
+}
+
+// ResolveProjectRef will resolve a ProjectRef to a Project, with the ProjectID.
```

Comment:
TBD if this should actually just return a ProjectRef, to follow the pattern we seem to have built since.  But no need to address in this PR  (unless you want to)

Reply:
I reverted it back, will address in another PR if needed.